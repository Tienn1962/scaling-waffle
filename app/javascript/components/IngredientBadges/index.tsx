import React from "react";

export default (props: { badges: string[] }) => {
  let badges = ["any"];

  if (props.badges.length > 0) {
    badges = props.badges;
  }

  return (
    <div className="flex flex-row items-center">
      <span>Ingredients :</span>
      {badges.map((badge) => (
        <div
          key={badge}
          className="badge bg-black"
        >
          {badge}
        </div>
      ))}
    </div>
  );
};
