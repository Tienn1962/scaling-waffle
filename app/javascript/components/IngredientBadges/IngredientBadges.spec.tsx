import React from "react";
import { fireEvent, render, screen } from "@testing-library/react";
import IngredientBadges from ".";

it("renders the badges", () => {
  const badges = ["bread", "chicken", "egg"];

  const component = render(<IngredientBadges badges={badges} />);

  expect(component).toMatchSnapshot();
});

it("renders a single badge with 'any' when passed an empty array", () => {
  const component = render(<IngredientBadges badges={[]} />);
  expect(component).toMatchSnapshot();
});
