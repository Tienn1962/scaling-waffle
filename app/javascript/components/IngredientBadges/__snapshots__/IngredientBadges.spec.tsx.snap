// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a single badge with 'any' when passed an empty array 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex flex-row items-center"
      >
        <span>
          Ingredients :
        </span>
        <div
          class="badge bg-black"
        >
          any
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex flex-row items-center"
    >
      <span>
        Ingredients :
      </span>
      <div
        class="badge bg-black"
      >
        any
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`renders the badges 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex flex-row items-center"
      >
        <span>
          Ingredients :
        </span>
        <div
          class="badge bg-black"
        >
          bread
        </div>
        <div
          class="badge bg-black"
        >
          chicken
        </div>
        <div
          class="badge bg-black"
        >
          egg
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex flex-row items-center"
    >
      <span>
        Ingredients :
      </span>
      <div
        class="badge bg-black"
      >
        bread
      </div>
      <div
        class="badge bg-black"
      >
        chicken
      </div>
      <div
        class="badge bg-black"
      >
        egg
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
